app/services/encrypted_attribute.rb

```diff
@@ -54,9 +52,14 @@ def try_decrypt(encryptor, key, cost)
   end
 
   def try_decrypt_with_all_keys(encryptor, cost)
-    encryption_keys.each do |key|
-      decrypted = try_decrypt(encryptor, key, cost) or next
-      return decrypted
+    encryption_keys.each_with_index do |key, index|
+      decrypted = try_decrypt(encryptor, key, cost)
+
+      return decrypted if decrypted
+
+      if index == encryption_keys.size - 1
```

Comment:
why check the index? if you raise *after* the `each` loop is done it does the same thing, more simply.
```ruby
encryption_keys.each do |key|
  decrypted = try_decrypt(encryptor, key, cost) or next
  return decrypted
end
raise Pii::EncryptionError, 'unable to decrypt attribute with any key'
```